pystac_monty/sources/gfd.py

```diff
@@ -0,0 +1,204 @@
+    hazard_profiles = MontyHazardProfiles()
+
+    def __init__(self, data: GFDDataSource):
+        self.data = data
+
+    def make_items(self) -> List[Item]:
+        """Create Items"""
+        items = []
+
+        event_items = self.make_source_event_items()
+        items.extend(event_items)
+
+        hazard_items = self.make_hazard_items()
+        items.extend(hazard_items)
+
+        impact_items = self.make_impact_items()
+        items.extend(impact_items)
+
+        return items
+
+    def get_event_collection(self, timeout: int = 30) -> Collection:
+        """Get Event collection"""
+        response = requests.get(self.gfd_events_collection_url, timeout=timeout)
+        collection_dict = json.loads(response.text)
+        return Collection.from_dict(collection_dict)
+
+    def get_hazard_collection(self, timeout: int = 30) -> Collection:
+        """Get Hazard collection"""
+        response = requests.get(self.gfd_hazards_collection_url, timeout=timeout)
+        collection_dict = json.loads(response.text)
+        return Collection.from_dict(collection_dict)
+
+    def get_impact_collection(self, timeout: int = 30) -> Collection:
+        """Get Impact collection"""
+        response = requests.get(self.gfd_impacts_collection_url, timeout=timeout)
+        collection_dict = json.loads(response.text)
+        return Collection.from_dict(collection_dict)
+
+    def _get_bounding_box(self, polygon: list):
+        """Get the bounding box from the polygon"""
+        lons, lats = zip(*polygon)  # Separate longitudes and latitudes
+        return [min(lons), min(lats), max(lons), max(lats)]
+
+    def make_source_event_items(self) -> List[Item]:
+        """Create the source event item"""
+        items = []
+        gfd_data = self.check_and_get_gfd_data()
+        if not gfd_data:
+            return []
+        for data in gfd_data:
+            item = self.make_source_event_item(data=data)
+            items.append(item)
+        return items
+
+    def make_source_event_item(self, data: dict) -> Item:
+        """Create the source event item"""
+
+        footprint = data["system:footprint"]
+        # Note: Convert LinearRing to Polygon as LinearRing is not supported in STAC spec.
+        geometry = {"type": "Polygon", "coordinates": [footprint["coordinates"]]}
+
+        description = data["dfo_main_cause"]
+
+        bbox = self._get_bounding_box(data["system:footprint"]["coordinates"])
+        # Episode number not in the source, so, set it to 1
+        episode_number = 1
+
+        startdate = pytz.utc.localize(datetime.fromtimestamp(data["system:time_start"] / 1000))
+        enddate = pytz.utc.localize(datetime.fromtimestamp(data["system:time_end"] / 1000))
+
+        item = Item(
+            id=f'{STAC_EVENT_ID_PREFIX}{data["system:index"]}',
```

Comment:
From the documentation, I think the id is unique and then sufficient to identity the event and consequant hazards and impacts
```suggestion
            id=f'{STAC_EVENT_ID_PREFIX}{data["id"]}',
```

Reply:
Done